oauth2_http/java/com/google/auth/oauth2/ExternalAccountCredentials.java

```diff
@@ -325,19 +326,30 @@ static ExternalAccountCredentials fromJson(
           /* scopes= */ null,
           /* environmentProvider= */ null);
     }
-    return new IdentityPoolCredentials(
-        transportFactory,
-        audience,
-        subjectTokenType,
-        tokenUrl,
-        new IdentityPoolCredentialSource(credentialSourceMap),
-        tokenInfoUrl,
-        serviceAccountImpersonationUrl,
-        quotaProjectId,
-        clientId,
-        clientSecret,
-        /* scopes= */ null,
-        /* environmentProvider= */ null);
+
+    // Validate that the userProject parameter, if present, is for a Workforce configuration.
+    // This doesn't validate the if the workforce audience is valid, only if it follows the
+    // workforce pattern.
+    Pattern workforceAudiencePattern =
```

Comment:
This should be within the relevant class right? Otherwise this check will not work if we create it via public constructor

Reply:
Yep - moved.